### PR TITLE
New liquid turret - Tsunami

### PR DESCRIPTION
**Pull request by Slava0135**
A lot of new units but only 2 new turrets. Wave is not particularly useful cos of it's range. It's still useful against fire but we need something better in late game.

### CODE REVIEW

**Comment by Slava0135** [comment]
oopsie forgot to add it to the research tree

File: core/src/mindustry/content/Blocks.java [hunk 1]
```diff
@@ -1458,6 +1458,26 @@ public void drawBase(Tile tile){}
             shootSound = Sounds.splash;
         }};
 
+        tsunami = new LiquidTurret("tsunami") {{
```
Comment (Quezler) [inline core/src/mindustry/content/Blocks.java:1461]:
```suggestion
        tsunami = new LiquidTurret("tsunami"){{
```

**Comment by Anuken** [comment]
- This turret should be 3x3, not 4x4. 4x4 is reserved only for the highest tier of combat turrets, and the sprite is not detailed enough to fit in with other 4x4 turrets.
- Missing a `-top` sprite for the liquid.
- Should be near the other 3x3 turrets, not next to the wave in building menu.
- Too cheap, should require a higher tier material like thorium at least.

**Comment by Slava0135** [comment]
Ok then. I'll try to fix the issues

**Comment by Slava0135** [comment]
Everything should be fine now. It's fine to me but you always can modify it

**Comment by Anuken** [comment]
Are you going to use EoD's sprite for this?

**Comment by Slava0135** [comment]
Not sure really. The problem with my sprite is that it has more thin lines compared to other turrets.

**Comment by Slava0135** [comment]
But I refused to add pipes to the sides as it looks weird to me.

**Comment by Anuken** [comment]
I prefer EoD's version; I think it's slightly cleaner and more consistent with the Wave sprite.

**Comment by Slava0135** [comment]
I feel the same but those pipes... idk about them

**Comment by Anuken** [comment]
They're fine.

**Comment by Slava0135** [comment]
Everything is ready I guess

**Comment by Slava0135** [comment]
oh no forgot comma in Blocks

**Comment by SnakkiZXZ** [comment]
as I see you cannot solve with the texture, so I specially drew this sprite for this, if there are 2 images.
tsunami
tsunami-liquid
Criticism is welcome.
P.S. you can add ```plastanium``` to craft, it is more expensive than ```titanium```.

**Comment by SnakkiZXZ** [comment]
oh no i am late?

**Comment by Slava0135** [comment]
too late but this looks more like a unit and not a turret

**Comment by SnakkiZXZ** [comment]
oh no

File: core/src/mindustry/content/Blocks.java [hunk 2]
```diff
@@ -1589,6 +1589,26 @@ public void drawBase(Tile tile){}
             reloadTime = 10f;
         }};
 
+        tsunami = new LiquidTurret("tsunami"){{
+            requirements(Category.turret, with(Items.metaglass, 100, Items.lead, 400, Items.titanium, 250, Items.thorium, 100));
+            ammo(
+                    Liquids.water, Bullets.waterShot,
```
Comment (Anuken) [inline core/src/mindustry/content/Blocks.java:1595]: Should use different bullet variants that deal more damage, draw larger and have more knockback.

File: core/src/mindustry/content/Blocks.java [hunk 3]
```diff
@@ -1589,6 +1589,26 @@ public void drawBase(Tile tile){}
...
+                    Liquids.slag, Bullets.slagShot,
+                    Liquids.cryofluid, Bullets.cryoShot,
+                    Liquids.oil, Bullets.oilShot
+            );
+            size = 3;
+            recoilAmount = 0f;
+            reloadTime = 1f;
```
Comment (Anuken) [inline core/src/mindustry/content/Blocks.java:1602]: This won't work. The reload should be at least 2-3; turrets can only shoot once per frame, so any FPS below 60 would make this turret fire slower than it should.

**Review by Anuken — CHANGES_REQUESTED** [review]

**Comment by Slava0135** [comment]
Made the changes

**Comment by Slava0135** [comment]
something went wrong and puddle/bullet size is so small

**Comment by Slava0135** [comment]
nvm everything is fine after repacking. The stats are somewhat reasonable

**Pull request merged**